include/dos_system.h

```diff
@@ -147,7 +147,54 @@ class DOS_File {
 	std::string name = {};
 	FlushTimeOnClose flush_time_on_close = FlushTimeOnClose::NoUpdate;
 	std::vector<FileRegionLock> region_locks = {};
-	/* Some Device Specific Stuff */
+
+	// The MediaDelayGuard class emulates media timing delays for file
+	// operations. It fixes games that either directly measure IO calls or
+	// are sensitive to IO calls that return too quickly. Examples are Deus
+	// (1996), Ishar 3 (1994), Robinson's Requiem (1994), and Time Warriors
+	// (1997). They (otherwise) report an iterrupt #00 handler error, which
+	// is triggered on a divide by zero.
+	//
+	// Create an object in DOS_File derived class's Read, Write, Seek, and
+	// Close functions. Add..() respective quantities. When the function
+	// goes out of scope, the cost for all added quantities will be applied.
+	//
+	class MediaDelayGuard {
```

Comment:
As I mentioned in my comment about the config setting's description further down, I would prefer `DiskAccessDelayGuard` and the term "disk access delay" in general. "Media" is very ambiguous (it could mean audio or video as well) and there's not much point in choosing a more ambiguous term when we have something much narrower that fits the bill, which is "DiskAccessDelay".

Reply:
How about this for the last paragraph:

```
Usage:

- Create `DiskAccessDelayGuard` object in `DOS_File` derived class's Read,
  Write, Seek, and Close functions.
- Call the `Add*()` methods to add the respective quantities.
- When the function (and thus the guard object) go out of scope, the cost for
  all added quantities will be applied.
```